### DISCUSSION TITLE
esp32 plug&pray external usb soundcart (I2S tda1541 DAC)

### DISCUSSION THREAD

**Discussion by OxF8** [pschatzmann/ESP32-A2DP#410] [Q&A] [answered]
Hello Dear, after brilliant work of the bluetooth->I2s(Philips) with the phone, I have pb with the PC - if seems breaks(something like echoes, or ticking with 0.5sec period, depends on the song) in the music. Probably it can be optimized(sizes, priorities?), but my ask is also for the USB plug&pray soundcard for the Win7/Win10. I prefer old Philips DACs on 44100/16bits and ESP32 works fine with them. Is there lazy way with Your soft to do it?

**Comment by pschatzmann** [comment, marked as answer]
I am sorry, but I only understand half of what you are saying.
In any case this is definitely not the right place for PC support! 
I haven't seen any windows pc for more then 20 years...

**Reply by OxF8** [reply]
:) & I remember 12 or 14 1.44 diskettes with funny operating system, called Linux :) Windows was some program under DOS(shit'n sugar mixed :) Before it, Apple ][ was with Monitor, to write the processor(6502) instructions directly :)  Ofcourse, UCSD Pascal was good. IBM360/370 with its true virtual machines operating system was something as elephant, which is unusable for the student :) No matter what is operating system or hardware - we must be able to solve tasks :) ok, will solve it in non-lazy way :)

**Reply by pschatzmann** [reply]
Just after university, at my first job I thought my work mate pulled a joke on me when he explained that to run a program on the IBM/370 I needed to write a program (in JCL)...

**Reply by OxF8** [reply]
Yes, JobControlLanguage needed for the older operating system - OS/DOS, but VM370 was true virtual machines operating system, with panel-based interface(today we call them windows;) There was also "light pen", and You can push with the pen on the texts, which are commands(today we call them touch-screen;) It was interesting realization - when You press with the pen on the screen, texts "flashed" one by one, while photo-diode of the pen received this flash. Depending on time passed, "pressed" text is determined.
Now about my task - I need ESP32-s2 or s3, to have usbAudio support. I don't want to go down to the firmware :)

**Reply by OxF8** [reply]
and every time devil tries to cheat us: https://github.com/sdima1357/esp32_usb_soft_host
